Task: base-entity

DangHuan2k3 opened the pull request:
## Description

- modified .prettierrc:tabwith
- added: entities
- updated: datatype for id and interval
- format: code base new .prettierrc
- created: DB on Cloud successfully

## Related Issues

## Checklist

- [x] Code follows the project’s code style guidelines.
- [x] Verified all related functionalities.
- [x] Updated documentation (if needed).

## Screenshots

![database on cloud](https://github.com/user-attachments/assets/11839dd0-1342-4c1f-be10-be9cee3b6b92)

## Additional Information

- Refs:
[one-to-one (typeorm)](https://orkhan.gitbook.io/typeorm/docs/one-to-one-relations)
[one-to-many (typeorm)](https://orkhan.gitbook.io/typeorm/docs/many-to-one-one-to-many-relations)
[many-to-many with properties (typeorm)](https://orkhan.gitbook.io/typeorm/docs/many-to-many-relations#many-to-many-relations-with-custom-properties)

DangHuan2k3 committed:
modified: user modules (controller, module, service)
modified: user.entity.ts
add: *.entity.ts
modified: .prettierrc tabwidth:4 -> tabwith:2

DangHuan2k3 committed:
add: unique constraint(s) for fields

DangHuan2k3 committed:
format: format code prettier

DangHuan2k3 committed:
Merge branch 'feature/base-entity' of https://github.com/Huy-Phan-159/Online-exam-with-notification-system into feature/base-entity

DangHuan2k3 committed:
modified: field about time of tables
removed: unneeded imports

DangHuan2k3 committed:
modified: field about time of tables
removed: unneeded imports
modified: enum constants

DangHuan2k3 committed:
sync: origin base-entity

DangHuan2k3 committed:
updated: base.entity id
modified: FK datatype  for relations many-to-many

NhatHuynh-Madison commented on src/configs/setup-swagger.ts:
```diff
@@ -4,7 +4,7 @@ import { NextFunction, Request, Response } from 'express';
   name: 'admin',
-  pass: 'admin',
+  pass: 'admin'
 };
```
move this credentials to env

NhatHuynh-Madison commented on src/main.ts:
```diff
@@ -20,11 +20,9 @@ async function bootstrap() {
+    })
   );
   await app.listen(port);
   console.info(`🚀 Server running on: http://localhost:${port}/api-docs`);
```
try logger

DangHuan2k3 replied:
i resolved that

NhatHuynh-Madison commented on src/middlewares/logging.middleware.ts:
```diff
@@ -3,8 +3,8 @@ import { Request, Response, NextFunction } from 'express';
-		next();
-	}
+  use(req: Request, res: Response, next: NextFunction) {
+    console.log(`Logging Request: ${req.method} ${req.url}`);
```
try logger

DangHuan2k3 committed:
add: field order in exams_questions
change: datatype id to UUID (crypto)
add: index for id for all entities, (firstName, lastName) for user entity
add: .env.example
add: apiDocumentationCredentials() in api-config.service.ts

DangHuan2k3 committed:
removed: field order in Question

DangHuan2k3 committed:
modified: name many-to-many relation tables
resolved: credentials swaagers

DangHuan2k3 committed:
removed: unnecessary lib(s)
